test/user.integration.js

```diff
@@ -178,6 +180,64 @@ describe('users - integration', function() {
           expect(injectedOptions).to.have.property('accessToken');
         });
     });
+
+    it('resets user\'s password', function() {
+      const User = app.models.User;
+      const credentials = {email: 'reset@example.com', password: 'pass'};
+      return User.create(credentials)
+        .then(u => {
+          this.user = u;
+          return triggerPasswordReset(credentials.email);
+        })
+        .then(info => {
+          return this.post('/api/users/reset-password')
+            .set('Authorization', info.accessToken.id)
+            .send({
+              newPassword: 'new password',
+            })
+            .expect(204);
+        })
+        .then(() => {
+          return User.findById(this.user.id);
+        })
+        .then(user => {
+          return user.hasPassword('new password');
+        })
+        .then(isMatch => expect(isMatch, 'user has new password').to.be.true());
+    });
+
+    it('rejects unauthenticated reset password request', function() {
+      return this.post('/api/users/reset-password')
+        .send({
+          newPassword: 'new password',
+        })
+        .expect(401);
+    });
+
+    it('injects reset password options from remoting context', function() {
+      const User = app.models.User;
```

Comment:
In the light of the discussion under https://github.com/strongloop/loopback/pull/3350#discussion_r110980156, maybe a better test name would be:

```js
it('configures "options" argument to be injected from remoting context'
```

Thoughts?